tests/tests_pytorch/models/test_amp.py

```diff
@@ -23,9 +24,26 @@
 from lightning_lite.plugins.environments import SLURMEnvironment
 from pytorch_lightning import Trainer
 from pytorch_lightning.demos.boring_classes import BoringModel, RandomDataset
+from pytorch_lightning.utilities.imports import _APEX_AVAILABLE
 from tests_pytorch.helpers.runif import RunIf
 
 
+@contextmanager
+def _suppress_apex_deprecation():
```

Comment:
You can remove all current test changes and just add a marker when runif is set with apex_amp

```diff
diff --git a/tests/tests_pytorch/conftest.py b/tests/tests_pytorch/conftest.py
index a8b54e737..ff1ddec48 100644
--- a/tests/tests_pytorch/conftest.py
+++ b/tests/tests_pytorch/conftest.py
@@ -289,6 +289,11 @@ def pytest_collection_modifyitems(items: List[pytest.Function], config: pytest.C
     for item in items:
         item.add_marker(deprecation_error)
 
+    apex_deprecation = pytest.mark.filterwarnings("ignore::apex.DeprecatedFeatureWarning")
+    for item in items:
+        if any(marker.name == "skipif" and marker.kwargs.get("amp_apex", False) for marker in item.own_markers):
+            item.add_marker(apex_deprecation)
+
 
 def pytest_addoption(parser):
     parser.addoption("--hpus", action="store", type=int, default=1, help="Number of hpus 1-8")
diff --git a/tests/tests_pytorch/helpers/runif.py b/tests/tests_pytorch/helpers/runif.py
index ae09f5b65..d25c246f8 100644
--- a/tests/tests_pytorch/helpers/runif.py
+++ b/tests/tests_pytorch/helpers/runif.py
@@ -153,6 +153,7 @@ class RunIf:
         if amp_apex:
             conditions.append(not _APEX_AVAILABLE)
             reasons.append("NVIDIA Apex")
+            kwargs["amp_apex"] = amp_apex
 
         if bf16_cuda:
             try:
```

I tested it locally and it works